code/modules/projectiles/projectile.dm

```diff
@@ -178,6 +187,32 @@
 				loc = A.loc
 			permutated.Add(A)
 			return 0
+		else if(!custom_impact)
+			var/obj/effect/overlay/beam/impact/impact = new(get_turf(src))

```

Comment:
overlay/beam are automatically qdel'd 1 second after being spawned, it's a relatively cheap object, like most effects.